src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -806,13 +809,59 @@ private static class ImageHelper implements ImageProvider {
         final boolean ignoreSigning;
         final Runtime.Version version;
         final Set<Archive> archives;
+        final Platform targetPlatform;
 
         ImageHelper(Configuration cf,
                     Map<String, Path> modsPaths,
                     ByteOrder order,
                     Path packagedModulesPath,
-                    boolean ignoreSigning) throws IOException {
-            this.order = order;
+                    boolean ignoreSigning,
+                    boolean verbose,
+                    PrintWriter log) throws IOException {
+            if (order != null) {
+                this.targetPlatform = Platform.runtime();
+                this.order = order;
+            } else {
+                Path javaBasePath = modsPaths.get("java.base");
+                assert javaBasePath != null : "java.base module path is missing";
+                if (this.isJavaBaseFromCurrentPlatform(javaBasePath)) {
+                    // this implies that the java.base module used for the target image
+                    // will correspond to the current platform. So this isn't an attempt to
+                    // build a cross-platform image. We use the current platform's endianness
+                    // in this case
+                    this.targetPlatform = Platform.runtime();
+                    this.order = ByteOrder.nativeOrder();
+                } else {
+                    // this is an attempt to build a cross-platform image. We now attempt to
+                    // find the target platform's arch and thus its endianness from the java.base
+                    // module's ModuleTarget attribute
+                    Optional<ResolvedModule> javaBase = cf.findModule("java.base");
+                    assert javaBase.isPresent() : "java.base module is missing";
+                    ModuleReference ref = javaBase.get().reference();
+                    String targetPlatformVal = null;
+                    if (ref instanceof ModuleReferenceImpl modRefImpl
+                            && modRefImpl.moduleTarget() != null) {
+                        targetPlatformVal = modRefImpl.moduleTarget().targetPlatform();
+                    }
+                    if (targetPlatformVal == null) {
+                        // could not determine target platform
+                        throw new IOException(
+                                taskHelper.getMessage("err.cannot.determine.target.platform"));
+                    }
+                    this.targetPlatform = Platform.parsePlatform(targetPlatformVal);
+                    this.order = this.targetPlatform.getNativeByteOrder();
+                    if (this.order == null) {
+                        // unsupported target platform
+                        throw new IOException(
+                                taskHelper.getMessage("err.unsupported.target.platform",
```

Comment:
IIUC, the current jlink implementation does not fail if the target platform is unknown as long as `--endian` option is specified.   So I think `--endian` may be useful to keep (rather than taking it out).   This error key should be `err.unknown.target.endianness` and the message can suggest to use `--endian` option to specify.

Reply:
I've updated the key to match what you suggest here.